src/intersections/rays.jl

```diff
@@ -205,4 +205,14 @@ function intersection(f, ray::Ray{3,T}, tri::Triangle{3,T}) where {T}
   return @IT Crossing ray(λ) f
 end
 
+function intersection(f, r::Ray{3,T}, p::Tetrahedron{3,T}) where {T}
+  mesh = boundary(p)
+  intersections = intersection.((r,), elements(mesh))
```

Comment:
Shouldn't the intersection also take into account the interior of the tetrahedron? If a ray passes through a tetrahedron, then we may get a segment? This intersection case is really difficult to express with our current representation. I know that Polyhedra.jl and other packages pursue other representations that facilitate some of these operations.

Reply:
Yeah I suppose I was considering the tetrahedron as the surfaces only. I come from the numerical PDEs world where you'd only really care about that. Help me consider the cases:

1. ray goes through volume creating a segment (this could also have further cases where if the segment endpoint is on a face, edge, or vertex. How specific do we want to be?)
2. ray colinear with edge
3. ray origin touches face, edge, vertex only
4. ?

Reply:
Let's start with a reduced list of cases, we don't need to differentiate the cases inside case (1), we just need to return the segment.

I am wondering how you would use this intersection result in the PDE world. Maybe that can guide us on the most appropriate cases.

Reply:
I suppose what I really meant was in numerical PDES (well, at least mesh-based schemes, especially FV), you only care about the faces and don't ever really consider the cell's volume (other than the literal value of the cell volume). So coming from this mindset (not necessarily right or wrong), naturally I'd expect/want an intersection like this to just give me the points the ray intersects the faces. If you were writing a simple toy FV code, you probably wouldn't be using this type of function at all anyways. Just trying to explain the mindset/expectations/though process, but it could be just me. Daniel VandenHeuvel could give a much better answer as he has developed some nice FV codes, my work is mainly in meshless PDE methods.

Reply:
Apologies for dragging my feet on this - I have been slammed busy lately.

Reply:
So what you are saying is that what you had in mind is really the intersection between the boundary of the tetrahedron and the ray. The boundary of the tetrahedron is a SimpleMesh, so maybe we need a method for Ray and SimpleMesh of Triangle instead?